OpenRA.Mods.Common/Warheads/HealthPercentageDamageWarhead.cs

```diff
@@ -24,32 +24,40 @@ public class HealthPercentageDamageWarhead : DamageWarhead
 		public override void DoImpact(WPos pos, Actor firedBy, IEnumerable<int> damageModifiers)
 		{
 			var world = firedBy.World;
+			var isRing = Spread.Length > 1 && Spread[1].Length > 0;
 
 			var debugVis = world.WorldActor.TraitOrDefault<DebugVisualizations>();
 			if (debugVis != null && debugVis.CombatGeometry)
 				world.WorldActor.Trait<WarheadDebugOverlay>().AddImpact(pos, Spread, DebugOverlayColor);
 
 			var range = Spread[0];
-			var hitActors = world.FindActorsInCircle(pos, range);
-			if (Spread.Length > 1 && Spread[1].Length > 0)
+
+			// FindActors*On*Circle is needed to include actors who are close enough to Spread[0] that their HitShape might intersect with it.
+			// However, we only want to exclude actors where CenterPosition is inside Range[1], so keeping *InCircle is correct there.
+			// TODO: Either properly account for hitshape intersection with Spread[1] instead, or drop ring support entirely.
+			// None of our shipping mods currently use this at time of writing.
```

Comment:
Note: If neither the TODO nor then 2nd option are good enough for this PR, I'll need help coming up with a way to check the hitshape intersection, as the check needs to work in reverse of what we normally do (here we need to check if the entire shape is *within* a radius), and I can't get my head around what the math would need to look like.

Reply:
Lets remove the ring code.  I don't understand why we had this in the first place.